### ISSUE TITLE
PointPillars validation results are too high

### ISSUE THREAD

**Issue by kvtw** [isl-org/Open3D-ML#346]
Is it possible there might be a bug in how the validation results are calculated? I downloaded the pretrained model of PointPillars (pytorch) and run the validation code, but the results are much higher than what it supposed to be. I used the original configs file for the pointpillars_kitti and able to reproduce the results on two different devices.

Environment:
- Ubuntu 18.04
- Open3D 0.13.0
- PyTorch 1.7.1
- CUDA 11.2 

Validation code:
```
import open3d.ml as _ml3d
import open3d.ml.torch as ml3d

cfg_file = #path_to_cfg
cfg = _ml3d.utils.Config.load_from_file(cfg_file)
dataset_kitti = ml3d.datasets.KITTI(dataset_path=#path_to_kitti_dataset, use_cache = True)
model = ml3d.models.PointPillars(**cfg.model)
pipeline_obj_det = ml3d.pipelines.ObjectDetection(model, dataset = dataset_kitti, **cfg.pipeline)
pipeline_obj_det.run_test()
```

Validation results: 
```
INFO - 2021-08-08 20:15:01,692 - object_detection - =============== mAP BEV ===============
INFO - 2021-08-08 20:15:01,692 - object_detection - class \ difficulty      0     1     2 
INFO - 2021-08-08 20:15:01,693 - object_detection - Pedestrian:          69.09 68.58 69.33 
INFO - 2021-08-08 20:15:01,693 - object_detection - Cyclist:             84.78 80.29 80.27 
INFO - 2021-08-08 20:15:01,693 - object_detection - Car:                 87.29 87.56 88.58 
INFO - 2021-08-08 20:15:01,693 - object_detection - Overall: 79.39
INFO - 2021-08-08 20:15:06,894 - object_detection - 
INFO - 2021-08-08 20:15:06,894 - object_detection - =============== mAP  3D ===============
INFO - 2021-08-08 20:15:06,894 - object_detection - class \ difficulty      0     1     2 
INFO - 2021-08-08 20:15:06,895 - object_detection - Pedestrian:          51.52 52.93 53.28 
INFO - 2021-08-08 20:15:06,895 - object_detection - Cyclist:             72.51 70.44 72.06 
INFO - 2021-08-08 20:15:06,895 - object_detection - Car:                 74.05 76.59 77.12 
INFO - 2021-08-08 20:15:06,895 - object_detection - Overall: 67.48
```
the moderate and hard results produced by the code are much higher than what is reported on the original paper. Additionally for the pedestrian and car, the 3D mAP results are higher for hard cases compared with the easy one, which is really unintuitive.

**Comment by kylevedder** [comment]
There are differences between the CVPR paper's setup and the setup in the `pointpillars_kitti` config; for example, the [initial learning rate and schedule](https://github.com/isl-org/Open3D-ML/blob/master/ml3d/configs/pointpillars_kitti.yml#L95) for Open3D-ML's implementation is different than [the first party config](https://github.com/nutonomy/second.pytorch/blob/master/second/configs/pointpillars/car/xyres_16.proto#L152), and I'd expect that there are other differences such as data augmentation.

Regarding the AP differences, those seem like they are fairly small. In KITTI the size of each pool (easy, medium, hard) varies, meaning that small amounts of deviation above the mean performance can be amplified due to the smaller dataset size.
